part-2.py

```diff
@@ -4,13 +4,40 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query, index=0):
+    # two base cases
+    if len(array) == index:
+        return False
+    elif array[index] == query:
+        return True
+    else:
+        return search(array,query, index+1)
 
 
 # is_palindrome
-
+def is_palindrome(text):
+    def helper(left, right):
+        if right - left <= 0:
+            return True
+        elif text[left]==text[right]:
+            return helper(left +1, right -1)
+        else:
+            return False
+    return helper(0, len(text) -1)  
 
 
 # digit_match
-
-
+def digit_match(left, right, matches=0):
+    if left == 0 and right ==0:
+        return matches + 1
+    elif left < 10 or right <10:
+        if left % 10 == right % 10:
+            return matches +1
```

Comment:
Consider if there's a way that you can DRY up this portion nothing that 0 % 10 = 0.

Nice work implementing this tricky problem!